crates/daphne/src/vdaf/prio3_latest.rs

```diff
@@ -0,0 +1,500 @@
+        agg_id: usize,
+        nonce: &[u8; 16],
+        public_share_data: &[u8],
+        input_share_data: &[u8],
+    ) -> Result<Prio3LatestPrepared<T, SEED_SIZE>, VdafError>
+    where
+        T: Type,
+        P: Xof<SEED_SIZE>,
+    {
+        // Parse the public share.
+        let public_share = Prio3PublicShare::get_decoded_with_param(&vdaf, public_share_data)?;
+
+        // Parse the input share.
+        let input_share =
+            Prio3InputShare::get_decoded_with_param(&(&vdaf, agg_id), input_share_data)?;
+
+        let binding = [CTX_STRING, &task_id.0].concat();
+        let ctx = binding.as_slice();
+
+        // Run the prepare-init algorithm, returning the initial state.
+        Ok(vdaf.prepare_init(
+            verify_key,
+            ctx,
+            agg_id,
+            &(),
+            nonce,
+            &public_share,
+            &input_share,
+        )?)
+    }
+}
+
+/// Consume the prep shares and return our output share and the prep message.
+pub(crate) fn prio3_prep_finish_from_shares(
+    config: &Prio3Config,
+    agg_id: usize,
+    task_id: TaskId,
+    host_state: VdafPrepState,
+    host_share: VdafPrepShare,
+    peer_share_data: &[u8],
+) -> Result<(VdafAggregateShare, Vec<u8>), VdafError> {
+    let (agg_share, outbound) = match (&config, host_state, host_share) {
+        (
+            Prio3Config::Count,
+            VdafPrepState::Prio3Field64(state),
+            VdafPrepShare::Prio3Field64(share),
+        ) => {
+            let vdaf = Prio3::new_count(2).map_err(|e| {
+                VdafError::Dap(
+                    fatal_error!(err = ?e, "failed to create prio3 count num_aggregators(2)"),
+                )
+            })?;
+            let (out_share, outbound) =
+                prep_finish_from_shares(&vdaf, task_id, agg_id, state, share, peer_share_data)?;
+            let agg_share = VdafAggregateShare::Field64(vdaf.aggregate(&(), [out_share])?);
+            (agg_share, outbound)
+        }
+        (
+            Prio3Config::Histogram {
+                length,
+                chunk_length,
+            },
+            VdafPrepState::Prio3Field128(state),
+            VdafPrepShare::Prio3Field128(share),
+        ) => {
+            let vdaf = Prio3::new_histogram(2, *length, *chunk_length)
+                .map_err(|e| VdafError::Dap(fatal_error!(err = ?e, "failed to create prio3 histogram from num_aggregators(2), length({length}), chunk_length({chunk_length})")))?;
+            let (out_share, outbound) =
+                prep_finish_from_shares(&vdaf, task_id, agg_id, state, share, peer_share_data)?;
+            let agg_share = VdafAggregateShare::Field128(vdaf.aggregate(&(), [out_share])?);
+            (agg_share, outbound)
+        }
+        (
+            Prio3Config::Sum { .. },
+            VdafPrepState::Prio3Field128(_),
+            VdafPrepShare::Prio3Field128(_),
+        ) => Err(VdafError::Dap(fatal_error!(err = "sum unimplemented")))?,
```

Comment:
For consistency with error handling elsewhere.
```suggestion
        ) => Err(VdafError::Dap(fatal_error!(err = "Prio3Sum is not supported in VDAF-13")))?,
```